docs/data-structures/next-greatest-to-right.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: Next Greatest Element to the Right
+slug: /data-structures/next-greatest-to-right
+---
+
+## Introduction
+
+Stack as a data structure has many applications, especially because of its 'Last In, First Out' principle. One of them is to find the greatest element to the right of each element in the array. 
```

Comment:
Add a background for the problem. Create another section to explain the problem so that the readers gets prompted to solve the question themselves.

Reply:
Done to the best of my ability.